Added `@` import to Webpack config, without needing a package

noviceGuru opened the pull request:
### Added `@` import
It is the default in `NextJS 13` and was not supported by storybook, while I was using it. The way I added it didn't need package to be installed or used

### Corrected the type error in `config.resolve`
Since `config.resolve` can be `undefined`, using `config.resolve.extensions` ran into `TypeScript` error. by adding a condition to the beginning of the line, I got rid of the `TS Error`.

### Worth to mention
I have a greenfield project in hand, using `NextJS13`. After adding `Storybook` for the first components, I ran into an error, that `Storybook` didn't recognize `@` imports that are default in `NextJS13`.
The solution in the documentation didn't work for me, while the one I added worked just fine.

noviceGuru committed:
config.resolve can be undefined by type

adding the condition, type error prevented

noviceGuru committed:
An easy way to configure @ import was added, which does not require any package

jonniebigodes commented on docs/snippets/common/storybook-main-ts-module-resolution.ts.mdx:
```diff
@@ -10,12 +10,14 @@ const config: StorybookConfig = {
-        extensions: config.resolve.extensions,
-      }),
-    ];
+    if(config.resolve){
```
@noviceGuru, mind adjusting this snippet's JavaScript version to factor in the same changes here for accuracy and consistency?

jonniebigodes commented on docs/builders/webpack.md:
```diff
@@ -128,6 +128,19 @@ Storybook's default Webpack configuration provides support for most project setu
 
 <!-- prettier-ignore-end -->
 
+While if you only want to add an import path, like `@` import which is used in `NextJS 13` default setup, you can use the following solution without adding a package:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-main-ts-module-resolution-atsign-import.js.mdx',
+    'common/storybook-main-ts-module-resolution-atsign-import.ts.mdx',
+  ]}****
+/>
+
+<!-- prettier-ignore-end -->
```
@noviceGuru we have a bit of room for improvement here, and we can adjust this to the following:
```md
However, if you're working with a framework that provides a default aliasing configuration (e.g., Next.js, Nuxt) and you want to configure Storybook to use the same aliases, you may not need to install any additional packages. Instead, you can extend the default configuration of Storybook to use the same aliases provided by the framework. For example, to set up an alias for the `@` import path, you can add the following to your `.storybook/main.js|ts` file:

<!-- prettier-ignore-start -->

<CodeSnippets
  paths={[
    'common/storybook-main-ts-module-resolution-atsign-import.js.mdx',
    'common/storybook-main-ts-module-resolution-atsign-import.ts.mdx',
  ]}
/>

<!-- prettier-ignore-end -->

```

jonniebigodes reviewed (changes requested):
@noviceGuru, thanks for taking the time to put together this pull request and help us improve the documentation. Appreciate it 🙏 ! From my end, I left a couple of items that require your attention so that we can merge this.

Additionally, while you're making the changes, do you mind checking the snippets, as they're failing the `pretty-docs` job to address it requires that you run the `yarn pretty-docs` script before committing.

Also, thanks for providing some additional context, and with that, I'm going to loop in a fellow maintainer that worked on the Next.js integration and see if this requires an additional pull request for the Next.Js framework documentation to factor in your issue. Sounds good?

@valentinpalkovic can you chime in on this when you have a moment? Appreciate it.

noviceGuru committed:
Description imporved, type correction applied to JS file

Description for aliasing configuration imporved, and type correction that was added to TS example, was added to JS version for accuracy and consistency.

noviceGuru committed:
pretty-docs script was run

jonniebigodes committed:
Merge branch 'next' into next

jonniebigodes commented on docs/builders/webpack.md:
```diff
@@ -128,6 +128,19 @@ Storybook's default Webpack configuration provides support for most project setu
+  paths={[
+    'common/storybook-main-ts-module-resolution-atsign-import.js.mdx',
+    'common/storybook-main-ts-module-resolution-atsign-import.ts.mdx',
+  ]}****
```
@noviceGuru can you remove the extra `****` here?

noviceGuru replied:
@jonniebigodes It's totally fine, sorry it slipped my hand, lost the notification among the emails.
I corrected it, and had to pull before pushing, so it should merge fast forward. Let me know for any other corrections.

jonniebigodes reviewed (changes requested):
@noviceGuru one small item for you that you left out. And apologize for not mentioning it earlier. Let me know once you've addressed it and I'll merge it.

noviceGuru committed:
typo corrected: extra stars in webpack.md deleted

noviceGuru committed:
Merge branch 'next' of github.com:noviceGuru/storybook into next

jonniebigodes committed:
Merge branch 'next' into next
